src/superagent-promise-plugin.js

```diff
@@ -3,8 +3,7 @@ function endPromise(req) {
 
   return new _Promise(function (resolve, reject) {
     req.end(function (err, res) {
-      err = err || res.error;
-      if (err) reject(err);
+      if (err || res.error) reject(res);
```

Comment:
hey @yachaka thanks for the PR. I think exposing the response object for rejected promises is a good idea. What do you think about putting the full response on the error object? That way there is no breaking changes and the failCallback can always expect an error object whether there is a thrown error or a response error.

Would something like this work?
```js
if (err) {
  reject(err); // res is undefined when err is defined
} else if (res.error) {
  res.error.response = res;
  reject(res.error); // res.error will have access to the full response object
} else {
  resolve(res);
}
```